addons/account/models/account_move.py

```diff
@@ -1376,9 +1380,9 @@ def create_exchange_rate_entry(self, aml_to_fix, amount_diff, diff_in_currency,
         for rec in self:
             if not rec.company_id.currency_exchange_journal_id:
                 raise UserError(_("You should configure the 'Exchange Rate Journal' in the accounting settings, to manage automatically the booking of accounting entries related to differences between exchange rates."))
-            if not rec.company_id.income_currency_exchange_account_id.id:
+            if not rec.company_id.accounting_company_id.income_currency_exchange_account_id.id:
                 raise UserError(_("You should configure the 'Gain Exchange Rate Account' in the accounting settings, to manage automatically the booking of accounting entries related to differences between exchange rates."))
-            if not rec.company_id.expense_currency_exchange_account_id.id:
+            if not rec.company_id.accounting_company_id.expense_currency_exchange_account_id.id:
```

Comment:
While accounts should be the same, a differentiation via the journal should be possible (as the new porter of the company attribute)